website/src/documentation/runtime/environments.md

```diff
@@ -0,0 +1,187 @@
+    </li>
+    <li>
+        Build a child image: <pre>docker build -f /path/to/new/Dockerfile</pre>
+    </li>
+</ol>
+
+### Modifying the original Dockerfile {#modifying-dockerfiles}
+
+1. Pull the [prebuilt SDK container image](https://console.cloud.google.com/gcr/images/apache-beam-testing/GLOBAL/beam/sdks/release) for your target language and version
+2. Customize the [Dockerfile](https://github.com/apache/beam/blob/master/sdks/python/container/Dockerfile). If you're adding dependencies from [PyPI](https://pypi.org/), use `base_image_requirements.txt` instead of the Dockerfile.
+3. [Reimage](#building-container-images) the container
+
+### Testing customized images
+
+Test customized images with PortableRunner. PortableRunner targets the default Docker images, like `${USER}-docker-apache.bintray.io/beam/python${version}:latest`. You can specify a different container with the `--environment_config` flag.
+
+#### Testing on local machines
+
+```
+python -m apache_beam.examples.wordcount --input /path/to/inputfile --output /path/to/write/counts \
+--runner=PortableRunner \
+--job_endpoint=embed \
+[--environment_config={docker_image_name}]
+```
+
+#### Testing on local Flink clusters
+
+<ol>
+    <li> Start a Flink job server on localhost:8099:
+        <pre>./gradlew :runners:flink:1.5:job-server:runShadow</pre>
+    </li>
+    <li> Run a pipeline on the Flink job server:
+<pre>python -m apache_beam.examples.wordcount --input /path/to/inputfile --output /path/to/write/counts \
+--runner PortableRunner \
+--job_endpoint localhost:8099 \
+[--environment_config {docker_image_name}]</pre>
+    </li>
+</ol>
+
+#### Testing on local Spark clusters
+
+<ol>
+    <li> Start a Spark job server on localhost:8099:
+        <pre>./gradlew :runners:spark:job-server:runShadow</pre>
+    </li>
+    <li> Run a pipeline on the Spark job server:
+<pre>python -m apache_beam.examples.wordcount --input /path/to/inputfile --output /path/to/write/counts \
+--runner=PortableRunner \
+--job_endpoint=localhost:8099 \
+[--environment_config={docker_image_name}]</pre>
+    </li>
+</ol>
+
+#### Testing on Google Cloud Dataflow
+
+```
+python -m apache_beam.examples.wordcount --input --input /path/to/inputfile --output /path/to/write/counts \
+--runner=DataflowRunner \
+--project={gcp_project_id} \
+--temp_location={gcs_location} \
+--worker_harness_container_image={docker_image_name} \
+--experiment=beam_fn_api \
+--sdk_location \ […]/beam/sdks/python/container/py{version}/build/target/apache-beam.tar.gz
+* version={2,35,36,37}
+```
+
+## Pushing container images
+
+After locally [building a container image](#building-container-images), push it to a remote Docker repository.
+
+Before pushing container images to your Docker registry, log in to your Bintray account with the `apache` Docker repository: ```docker login```
```

Comment:
Pushing images to a remote repository is an option for users. Pushing to which repository is also user's choice, we can give Bintray as an example.